trio/_core/tests/test_run.py

```diff
@@ -840,6 +840,16 @@ def test_cancel_points():
             with pytest.raises(_core.Cancelled):
                 await _core.checkpoint()
 
+    # change shielding while a cancellation is in flight
```

Comment:
would it be difficult to make this a separate test?

Testing is hard and itself error-prone, so there's some advantage to testing one thing at a time.  Also some opportunity for self-description via test function name.